### PR TITLE
Epub: when relocating to the last page, ensure we put the slider at 100

### LINKED ISSUE #8309: Investigate potential bug with progress/completion
There may be a bug in develop with progress/completion. It seems like there are mismatches between the progress that is displayed in the UI and the progress that is calculated, and that some resources cannot be successfully "completed". So far, it seems to be occurring in the epub content renderer, and possibly with videos. 

The purpose of this issue is to investigate and see if the issue is reproducible, and investigate where the problem might be happening, to create a detailed follow up issue.

image

### PR DESCRIPTION
**Pull request by nucleogenesis**
## Summary

Previously we always used the `location.start.percentage` value in the Epub renderer to set the slider's percentage. This ensures that when we are moving to a location whose `location.end.percentage` is 100, we set the slider value to 100 (completing the resource).

## References

Fixes #8309 

## Reviewer guidance

Does it fix it?

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by rtibbles** [comment]
Flagging that resource completion is based on 'visiting every page' now, and not on the slider location being at 100 percent. So while this might be a useful fix, I'm not sure it will fix the underlying issue.

File: kolibri/plugins/epub_viewer/assets/src/views/EpubRendererIndex.vue [hunk 1]
```diff
@@ -704,7 +704,13 @@
         this.currentSection = this.getCurrentSection(currentLocationStart);
       },
       relocatedHandler(location) {
-        this.sliderValue = location.start.percentage * 100;
+        // Ensures that when we're on the last page, we set the slider value to 100
+        // otherwise, we show the slider % using the start
+        if (location.end.percentage === 100) {
```
Comment (rtibbles) [inline kolibri/plugins/epub_viewer/assets/src/views/EpubRendererIndex.vue]: Should this be `== 100` or `== 1` - I am guessing based on the fact that we multiply it by `100` below that the `location.end.percentage` value is actually a fraction between 0 and 1?

Comment (nucleogenesis) [reply kolibri/plugins/epub_viewer/assets/src/views/EpubRendererIndex.vue]: Good catch. After looking at what I get in the `location` object, apparently it puts a `atEnd: true` bool flag on it when it's the last page, so I refactored to use that.

**Comment by nucleogenesis** [comment]
> Flagging that resource completion is based on 'visiting every page' now, and not on the slider location being at 100 percent. So while this might be a useful fix, I'm not sure it will fix the underlying issue.

Figured it out :)

**Comment by marcellamaki** [comment]
Hi @nucleogenesis - so I do think what you've described this fix as doing is accurate and working. My question is kind of the same as Richard's which is.... is there maybe something else going on with epub progress that deserves more investigation? I don't think there's any downside to merging this, but I did some more "realistic" manual QA that I usually do and actually read through an epub (and I picked one in French, and my French is terrible, so that it would be an authentic learning experience and not me skimming and clicking through 🙃). Despite actually fully engaging with it _twice_, it still shows as "in progress" and every time I open it, it comes up to 60% (and the corresponding page.) Again - this is sort of outside what you have described fixing here, but it does seem like we should attend to it. Maybe I can create a follow up issue for this, and add a bit more context?

**Comment by rtibbles** [comment]
I think we can merge this, but not mark the issue as fixed, seems easiest.

**Review by marcellamaki — APPROVED** [review]

**Comment by pcenov** [comment]
Hi everyone, today I had a look at this [build](https://buildkite.com/learningequality/kolibri-windows/builds/1773) and just want to add that:
1. The slider goes to 100% even before reaching the last page which is confusing from an end-user point of view.
2. Out of a number of epub resources that I have attempted to complete only one was marked as completed (Tinisiima-The_Ungrateful_Man_ePub_Document) but that happened while I was at about 80% of the book, which is again quite confusing. I couldn't achieve getting any of the other epub files to be marked as completed.




@radinamatic

**Pull request merged**